File: src/test/java/org/stt/importer/STTItemReaderTest.java
package org.stt.importer;

import org.hamcrest.Matchers;
import org.junit.Assert;
import org.junit.Test;
import org.stt.model.TimeTrackingItem;
import org.stt.persistence.ItemReader;
import org.stt.persistence.stt.STTItemReader;

import java.io.StringReader;
import java.time.LocalDateTime;
import java.util.Optional;

public class STTItemReaderTest {

	@Test
	public void multiLineCommentGetsImportedCorrectly() {

		// GIVEN
		StringReader stringReader = new StringReader(
				"2012-10-10_22:00:00 2012-11-10_22:00:01 this is\\n a multiline\\r string\\r\\n with different separators");
		ItemReader theReader = new STTItemReader(stringReader);

		// WHEN
		Optional<TimeTrackingItem> readItem = theReader.read();

		// THEN
		Assert.assertEquals(
				"this is\n a multiline\r string\r\n with different separators",
                readItem.get().getActivity());
    }

	@Test
	public void onlyStartTimeGiven() {

		// GIVEN
		StringReader stringReader = new StringReader("2012-10-10_22:00:00");
		ItemReader theReader = new STTItemReader(stringReader);

		// WHEN
		Optional<TimeTrackingItem> readItem = theReader.read();

		// THEN
        LocalDateTime time = LocalDateTime.of(2012, 10, 10, 22, 00, 00);
        Assert.assertThat(time, Matchers.equalTo(readItem.get().getStart()));
	}

	@Test
	public void startTimeAndCommentGiven() {

		// GIVEN
		StringReader stringReader = new StringReader(
				"2012-10-10_22:00:00 the long comment");
		ItemReader theReader = new STTItemReader(stringReader);

		// WHEN
		Optional<TimeTrackingItem> readItem = theReader.read();

		// THEN
        LocalDateTime time = LocalDateTime.of(2012, 10, 10, 22, 00, 00);
        Assert.assertThat(time, Matchers.equalTo(readItem.get().getStart()));
		Assert.assertThat("the long comment",
                Matchers.equalTo(readItem.get().getActivity()));
    }
}

File: src/test/kotlin/org/stt/importer/STTItemReaderTest.kt
package org.stt.importer

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.stt.persistence.stt.STTItemReader
import java.io.StringReader
import java.time.LocalDateTime

class STTItemReaderTest {

    @Test
    fun multiLineCommentGetsImportedCorrectly() {

        // GIVEN
        val stringReader = StringReader(
                "2012-10-10_22:00:00 2012-11-10_22:00:01 this is\\n a multiline\\n string\\n")
        val theReader = STTItemReader(stringReader)

        // WHEN
        val readItem = theReader.read()

        // THEN
        assertThat("this is\n a multiline\n string\n").isEqualTo(readItem!!.activity)
    }

    @Test
    fun onlyStartTimeGiven() {

        // GIVEN
        val stringReader = StringReader("2012-10-10_22:00:00")
        val theReader = STTItemReader(stringReader)

        // WHEN
        val readItem = theReader.read()

        // THEN
        val time = LocalDateTime.of(2012, 10, 10, 22, 0, 0)
        assertThat(time).isEqualTo(readItem!!.start)
    }

    @Test
    fun startTimeAndCommentGiven() {

        // GIVEN
        val stringReader = StringReader(
                "2012-10-10_22:00:00 the long comment")
        val theReader = STTItemReader(stringReader)

        // WHEN
        val readItem = theReader.read()

        // THEN
        val time = LocalDateTime.of(2012, 10, 10, 22, 0, 0)
        assertThat(time).isEqualTo(readItem!!.start)
        assertThat("the long comment").isEqualTo(readItem.activity)
    }
}
